generate_tests.sh

```diff
@@ -0,0 +1,201 @@
+RUST_TOP=/home/maria/builds/RV/rust
+RUST_TESTS=${RUST_TOP}/tests
+
+# Top directory of the mir semantics test repository.
+MIRSEMANTICS_COMPILETEST=$(pwd)
+
+# Choose whether to run the make step to generate *.mir, *.stdout, *.stderr.
+# true to run
+rebuild=true
+# Choose whether to include fixed tests (that include the run-rustfix header).
+# true to include
+remove_fixed=false
+# Choose whether to include tests with no main or an empty main function.
+# true to include
+remove_no_main=false
+# Choose whether to include tests that do not include the run-pass header.
+# true to include
+remove_no_runpass=false
+# Choose whether to write the list of identified tests (.rs) and generated IR
+# (.mir) and output files(.stdout, .stderr) to files.
+# true to create the files containing these lists.
+opt_output=true
+# Choose whether to copy the identified/generated files to the mir semantics
+# test repository. true to copy.
+cpflag=false
+# Choose whether to stage the new/modified files for commit to the MIR
+# semantics test repository. If true, it supersedes the cpflag.
+# true to copy and stage them for commit.
+addflag=false
+
+# Command line arguments can be used to modify these default values.
+print_usage() {
+  printf "Usage: $(basename $0) [-b] [-f] [-m] [-p] [-s] [-c] [-a] [-h]\n\n"
+  printf "options:\n"
+  printf "a\tCopy the new and modified files to the MIR semantics test repository, and stage them for commit.\n"
+  printf "c\tCopy the new and modified files to the MIR semantics test repository. Do not stage them for commit.\n"
+  printf "b\tSkip the make step. Assumes that the *.mir, *.stdout, *.stderr files are available.\n"
+  printf "f\tSkip fixed tests (that include the run-rustfix header).\n"
+  printf "m\tSkip tests that have an empty or missing main function.\n"
+  printf "p\tSkip tests that do not have the run-pass header.\n"
+  printf "s\tSkip writing the list of identified tests and generated files out to files.\n"
+  printf "h\tPrint this usage and exit.\n"
+}
+
+while getopts 'bfmpscah' opt; do
+  case "${opt}" in
+    b) rebuild='false' ;;
+    f) remove_fixed='true' ;;
+    m) remove_no_main='true' ;;
+    p) remove_no_runpass='true' ;;
+    s) opt_output='false' ;;
+    c) cpflag='true' ;;
+    a) addflag='true' ;;
+    ?|h) print_usage
+         exit 1 ;;
+  esac
+done
+
+echo "cd ${RUST_TESTS}"
+cd ${RUST_TESTS}
+
+echo "MAKE STEP"
+if [ "$rebuild" = true ]; then
+  # Copy the Makefile from the mir semantics test repository.
+  echo "cp ${MIRSEMANTICS_COMPILETEST}/Makefile ${RUST_TESTS}"
+  cp ${MIRSEMANTICS_COMPILETEST}/Makefile ${RUST_TESTS}
+
+  echo "make clean"
+  make clean
+  echo "make ui-mir"
+  make ui-mir
+  echo "make ui-out"
+  make ui-out
+fi
+
+echo "INITIAL INPUT SELECTION"
+# Find all generated mir files
+UI_MIR_GEN=`find ui -name '*.mir'`
+# Select the source Rust files for those
+UI_RS=""
+for file in $UI_MIR_GEN; do
+    src_file=${file%.mir}.rs
+    UI_RS="$UI_RS $src_file"
```

Comment:
I just wanted to make some quick points here about some best practices for shell scripts (as far as I understand them --- though note that I am far from a shell expert):

1. This idiom relies on the paths that are being concatenated do not contain any spaces. On Linux, this is generally true --- but Mac/Windows doesn't always cooperate here.

2. When applied to very large inputs, this idiom can result in variable expansions which exceed the default maximum command length. To avoid this, one can write the inputs to a file and then use the `xargs` tool to construct commands for smaller groups of input lines (or just a single line).

Since this script is for internal, controlled use, I don't think the above two points apply really, but I wanted to share as a future reference point.